replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/QueryServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+package org.codice.ditto.replication.api.impl;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.stream.Stream;
+import org.apache.commons.io.IOUtils;
+import org.codice.ditto.replication.api.QueryService;
+import org.codice.ditto.replication.api.ReplicationException;
+import org.codice.ditto.replication.api.data.Query;
+import org.opengis.filter.Filter;
+
+/** A service which fetches query metacards from the catalog and returns them as {@link Query}s. */
+public class QueryServiceImpl implements QueryService {
+
+  private final CatalogFramework framework;
+
+  private final FilterBuilder builder;
+
+  private final InputTransformer xmlTransformer;
+
+  /**
+   * Creates a new QueryService
+   *
+   * @param framework The catalog framework to retrieve metacards from.
+   * @param builder A filter builder
+   * @param xmlTransformer An xml input transformer for transforming the query metacard xml on a
+   *     workspace metacard into a {@link Metacard} object.
+   */
+  public QueryServiceImpl(
+      CatalogFramework framework, FilterBuilder builder, InputTransformer xmlTransformer) {
+    this.framework = framework;
+    this.builder = builder;
+    this.xmlTransformer = xmlTransformer;
+  }
+
+  /**
+   * Provides a stream for retrieving all saved user queries as {@link Query}s.
+   *
+   * @return all the user queries saved on the system
+   */
+  @Override
+  public Stream<Query> queries() {
+    Filter filter = builder.attribute(Core.METACARD_TAGS).is().equalTo().text("workspace");
+    QueryImpl query = new QueryImpl(filter);
+    query.setPageSize(100);
```

Comment:
❓ Will we only ever return the first 100 workspaces? Could that be a problem?

Reply:
Yea we only return the first 100 workspaces. I think that it's reasonable to expect one user to not have more that 100 workspaces but could see this being an issue if, say, multiple people shared one set of credentials. Then there's a good chance they will have more that 100. We could return a resultIterable but then we would need to incorporate paging into the query selector. Or we could only fetch queries based on what the user types into the search box, but I think it would be difficult to search based on query titles since they are saved as metacard xml on workspace metacards.

Reply:
I thought there was a notion of shared workspaces. If that is the case and if the list be greater than 100, how do we know that the first 100 are the right one to display?

Reply:
I feel like 98% of the time this will be sufficient. I don't think we should hold this pr up for this reason. If needed we can iterate on some kind of paging. We could also add a sort to our query so the most recent 100 workspaces are returned with would probably make it work 99% of the time. (like how I am just making up %)

Reply:
I would go for the last thing you suggested if it is easy.

Reply:
I'll add a sort based on the modified date. That should help.